### PR TITLE
Shimmers in Advances and Team-advances

### PR DESCRIPTION
**Pull request by siddharth0-0**

### CODE REVIEW

File: src/app/fyle/team-advance/team-advance.page.ts [hunk 1]
```diff
@@ -106,7 +104,11 @@ export class TeamAdvancePage implements OnInit {
     );
 
     this.loadData$.subscribe(noop);
-    this.teamAdvancerequests$.subscribe(noop);
+    this.teamAdvancerequests$.subscribe((res) => {
+      if (res) {
+        this.isLoading = false;
```
Comment (iravisankar) [inline src/app/fyle/team-advance/team-advance.page.ts:109]: use a pipe

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 2]
```diff
@@ -147,64 +146,63 @@ export class MyAdvancesPage {
...
+              });
+            }
+            newArr = this.utilityService.sortAllAdvances(filters.sortDir, filters.sortParam, newArr);
+            return newArr;
+          })
         )
       )
     );
+
+    this.advances$.subscribe((res) => {
```
Comment (iravisankar) [inline src/app/fyle/my-advances/my-advances.page.ts:201]: use a pipe for this..

**Review by iravisankar — APPROVED** [review]

**Review by Dimple16 — APPROVED** [review]

File: src/app/fyle/team-advance/team-advance.page.ts [hunk 3]
```diff
@@ -69,7 +67,12 @@ export class TeamAdvancePage implements OnInit {
         }
         return acc.concat(curr);
       }, [] as ExtendedAdvanceRequest[]),
-      shareReplay(1)
+      shareReplay(1),
+      tap((res) => {
```
Comment (Julias0) [inline src/app/fyle/team-advance/team-advance.page.ts:71]: Put this inside `finalize`

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 4]
```diff
@@ -147,63 +146,61 @@ export class MyAdvancesPage {
...
-      )
+      ),
+      switchMap((advArray) =>
+        //piping through filterParams so that filtering and sorting happens whenever we call next() on filterParams
+        this.filterParams$.pipe(
+          map((filters) => {
+            let newArr = cloneDeep(advArray);
+
+            if (filters && filters.state && filters.state.length > 0) {
+              newArr = advArray.filter((adv) => {
```
Comment (Julias0) [inline src/app/fyle/my-advances/my-advances.page.ts:179]: extract out the contents of filter into a separate function

File: src/app/fyle/my-advances/my-advances.page.ts [hunk 5]
```diff
@@ -147,63 +146,61 @@ export class MyAdvancesPage {
...
+          }),
+          sortResults
         )
-      )
+      ),
+      switchMap((advArray) =>
+        //piping through filterParams so that filtering and sorting happens whenever we call next() on filterParams
+        this.filterParams$.pipe(
+          map((filters) => {
+            let newArr = cloneDeep(advArray);
```
Comment (Julias0) [inline src/app/fyle/my-advances/my-advances.page.ts:176]: newArr -> advArrayCopy

avoid names like arr etc

**Review by Julias0 — APPROVED** [review]

**Pull request merged**